### PR TITLE
notebooks - add args + promote some args to GA

### PR DESCRIPTION
**Pull request by DrFaust92**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
notebooks: promoted `nicType` and `reservationAffinity` in `google_notebooks_instance` to GA
```

```release-note:enhancement
notebooks: added `bootDiskType` support for `PD_EXTREME` in `google_notebooks_instance`
```

```release-note:enhancement
notebooks: added `softwareConfig.upgradeable, `softwareConfig.postStartupScriptBehavior`, `softwareConfig.kernels` in `google_notebooks_runtime`
```

### CODE REVIEW

File: mmv1/products/notebooks/api.yaml [hunk 1]
```diff
@@ -841,12 +850,16 @@ objects:
         description: |
           The config settings for accessing runtime.
         properties:
-          - !ruby/object:Api::Type::String
+          - !ruby/object:Api::Type::Enum
```
Comment (c2thorn) [inline mmv1/products/notebooks/api.yaml:853]: Why update this from string to enum? We have been tending to do the opposite so we don't have to keep up with the API on what's actually validated. 
I'd be very hesitant to approve String -> Enum without strong reasoning.

**Review by c2thorn — CHANGES_REQUESTED** [review]
Hi @DrFaust92,

Can we get examples/tests using the added fields?

File: mmv1/products/notebooks/api.yaml (hunk 1 above)
Comment (DrFaust92) [reply mmv1/products/notebooks/api.yaml:853]: i can revert but plan time validations are helpful when API (more often than not...) is vague in its response for bad input. lmk what you decide

**Comment by DrFaust92** [comment]
ill add relevant tests

File: mmv1/products/notebooks/api.yaml (hunk 1 above)
Comment (c2thorn) [reply mmv1/products/notebooks/api.yaml:853]: That's understandable. We've run into a fair amount of complaints when the validation is over-restrictive, mostly due to the API enabling new values and the provider being too slow in updating.
Removing the helpfulness of plan-time validations in exchange for reduction of future maintenance is the direction our team has been heading, so I'll go with with reverting at this time.

Comment (DrFaust92) [reply mmv1/products/notebooks/api.yaml]: reverted

**Comment by DrFaust92** [comment]
added tests to new fields

**Review by c2thorn — APPROVED** [review]

**Pull request merged**